generate.sh

```diff
@@ -98,97 +98,110 @@ fi
 
 mkdir yaydoctemp
 BUILD_DIR=$(pwd)/yaydoctemp
+TEMP_NAME=yaydoctemp
+function generate {
 
-cd $ROOT_DIR
+  cd $1
```

Comment:
i think we should use all the variables like `${variable}` instead of `$variable`
@pri22296  suggestions?